drivers/syslog/ramlog.c

```diff
@@ -267,13 +266,24 @@ static void ramlog_initbuf(void)
 #endif
 
 /****************************************************************************
- * Name: ramlog_addchar
+ * Name: ramlog_addbuf
  ****************************************************************************/
 
-static int ramlog_addchar(FAR struct ramlog_dev_s *priv, char ch)
+static ssize_t ramlog_addbuf(FAR struct ramlog_dev_s *priv,
+                             FAR const char *buffer, size_t len)
 {
+  int readers_waken;
   irqstate_t flags;
-  size_t nexthead;
+  int space;
+  int remain;
+  int head;
+  int ret;
+
+  ret = nxmutex_lock(&priv->rl_lock);
```

Comment:
why add mutex here? ramlog_addbuf must be callable in the interrupt context.

Reply:
I have a question about this, but I haven't changed it here, and it was included in the original addbuf.
In the original addbuf, both nxmutex_lock and enter_critical_section are used. Is this necessary?
I don't quite understand why the protection needs to be added twice.

Reply:
ramlog_addchar may be called in the interrupt context. Since ramlog_addchar is implemented on top of ramlog_addbuf, ramlog_addbuf need support the interrupt context too.

Reply:
You are right, there will be a problem when len>2*rl_bufsize